### PR TITLE
[Merged by Bors] - feat(CategoryTheory): the category of commutative squares

### PR DESCRIPTION
**Pull request by joelriou**
In this PR, we define a bundled version of `CommSq` which allows to consider commutative squares as objects in a category `Square C`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Square.lean
Comment (erdOne) [inline Mathlib/CategoryTheory/Square.lean:95]:
```suggestion
```

File: Mathlib/CategoryTheory/Square.lean
Comment (erdOne) [inline Mathlib/CategoryTheory/Square.lean:22]: This diagram looks a bit off in both github and my vscode. Maybe use `->` arrows to avoid kerning issues?

File: Mathlib/CategoryTheory/Square.lean
Comment (erdOne) [inline Mathlib/CategoryTheory/Square.lean:152]:
```suggestion
      comm₂₄ := by simp only [← cancel_mono e₄.hom, assoc, Iso.inv_hom_id,
                      comp_id, comm₂₄, Iso.inv_hom_id_assoc]
      comm₃₄ := by simp only [← cancel_mono e₄.hom, assoc, Iso.inv_hom_id,
```

**Review by erdOne — APPROVED** [review]
I was just thinking about suggesting the same thing when I saw the other PR.
LGTM modulo minor comments.

**Comment by joelriou** [comment]
Thanks for the reviews @erdOne! Indeed, while finding the better way to formalize Mayer-Vietoris squares, it has become more obvious to me that introducing this category `Square C` was the better approach.

**Comment by erdOne** [comment]
Thanks!
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**
